jupyterlab_chameleon/artifact.py

```diff
@@ -373,6 +403,24 @@ def list(self) -> dict:
         self.log.info(f"Fetched {len(artifacts)} artifacts.")
         return artifacts
 
+    def metric(self, uuid: str, slug: str, metric: str, origin:str):
+        prepared_req = self.prepare_metric(uuid, slug)
+        metric_url = prepared_req.get("url")
+        metric_method = prepared_req.get("method", "PUT")
+        metric_headers = prepared_req.get("headers", {})
+
+        if not metric_url:
+            raise ValueError("Malformed patch request")
+
+        res = requests.request(
+            url=metric_url,
+            method=metric_method,
+            headers=metric_headers,
+            json={"metric": metric,
+                  "origin": origin},
```

Comment:
This call also requires the `write_metrics` scope on its token. The `write_metrics` scope is also supposed to be limited to admins. I think this request has to be made by hub rather than lab, since hub has admin credentials and lab does not. Hub would obtain a trovi token with `write_metrics` scope, and make a request to update the metrics on behalf of the user (which is why we have the `origin` parameter, so we know what user initiated the call).

If this call works from lab, then there is a security issue with Trovi. The code which [acquires the Trovi token](https://github.com/ChameleonCloud/jupyterlab-chameleon/blob/f39c797652d2dc1a83d114829d866bcd461b6485/jupyterlab_chameleon/trovi.py#L55) does not request the required scope.